### PR TITLE
DFPL - 1278 Notifications in list gatekeeping hearing

### PR DESCRIPTION
**Pull request by KatKovacs1**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFPL-1278

### Change description ###

Sending out notifications and SDO on the back of list gatekeeping hearing event

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ListGatekeepingHearingController.java [hunk 1]
```diff
@@ -100,7 +105,6 @@ public CallbackResponse handleAboutToStart(@RequestBody CallbackRequest callback
         }
 
         caseDetails.getData().putAll(hearingsService.populateHearingLists(caseData));
-        caseDetails.getData().put("sendNoticeOfHearing", YES.getValue());
```
Comment (benouaer) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ListGatekeepingHearingController.java:103]: we can keep this here, this should be handled through 1201 now

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ListGatekeepingHearingController.java [hunk 2]
```diff
@@ -111,6 +115,7 @@ public CallbackResponse handleAboutToStart(@RequestBody CallbackRequest callback
     public CallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
 
         final CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        caseDetails.getData().put("sendNoticeOfHearing", YES.getValue());
```
Comment (benouaer) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ListGatekeepingHearingController.java:118]: no need to move this here, it's being set through a readonly field now

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/listgatekeepinghearing/ListGatekeepingHearingControllerSubmittedTest.java [hunk 3]
```diff
@@ -1,207 +1,324 @@
...
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testAddress;
 import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocmosisDocument;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocument;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocumentBinary;
 import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocumentReference;
 
 @WebMvcTest(ListGatekeepingHearingController.class)
 @OverrideAutoConfiguration(enabled = true)
 @DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
 class ListGatekeepingHearingControllerSubmittedTest extends AbstractCallbackTest {
```
Comment (benouaer) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/listgatekeepinghearing/ListGatekeepingHearingControllerSubmittedTest.java:95]: there's already a ListGatekeepingHearingControllerSubmittedTest as part of 1201, so this one can be removed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java [hunk 4]
```diff
@@ -32,8 +33,9 @@ public Optional<GatekeepingOrderEvent> buildEventToPublish(CaseData caseData, St
         if (null != sdo.getOrderDoc()) {
             event.order(sdo.getOrderDoc());
             event.languageTranslationRequirement(sdo.getTranslationRequirements());
-            // if we are in the gatekeeping state send the NoP related notifications
-            event.notificationGroup(isInGatekeeping(previousState) ? SDO_AND_NOP : SDO);
+            // if we are in the gatekeeping or gatekeeeping-listing state send the NoP related notifications
+            event.notificationGroup((isInGatekeeping(previousState) || isInGatekeepingListing(previousState))
```
Comment (benouaer) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java:37]: is this notification ever gonna be sent during gatekeeping state?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java [hunk 5]
```diff
@@ -107,6 +108,10 @@ public static boolean isInGatekeepingState(CaseDetails caseDetails) {
         return isInState(GATEKEEPING, caseDetails);
     }
 
+    public static boolean isInGatekeepingListingState(CaseDetails caseDetails) {
```
Comment (benouaer) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java:111]: don't really see the value in this over isInState

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java [hunk 6]
```diff
@@ -48,4 +50,8 @@ public Optional<GatekeepingOrderEvent> buildEventToPublish(CaseData caseData, St
     private boolean isInGatekeeping(State previousState) {
         return GATEKEEPING == previousState;
     }
+
+    private boolean isInGatekeepingListing(State previousState) {
+        return GATEKEEPING_LISTING == previousState;
```
Comment (tomxelliott) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java:55]:
```suggestion
        return GATEKEEPING_LISTING.equals(previousState);
```

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/listgatekeepinghearing/ListGatekeepingHearingControllerSubmittedTest.java (hunk 3 above)
Comment (mapingo) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/listgatekeepinghearing/ListGatekeepingHearingControllerSubmittedTest.java:95]: The one in 1201 was not working, while this one works after the changes applied by Katalin

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java (hunk 4 above)
Comment (mapingo) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java:37]: Removing check for gatekeeping state

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java (hunk 6 above)
Comment (mapingo) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/GatekeepingOrderEventNotificationDecider.java:55]: Done

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java (hunk 5 above)
Comment (mapingo) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDetailsHelper.java:111]: Removed

**Review by tomxelliott — APPROVED** [review]

**Review by benouaer — APPROVED** [review]

**Review by benouaer — APPROVED** [review]

**Pull request merged**